util/secscan/v4/api.py

```diff
@@ -108,16 +111,24 @@ def index(self, manifest, layers):
         # TODO: Better method of determining if a manifest can be indexed (new field or content-type whitelist)
         assert isinstance(manifest, ManifestDataType) and not manifest.is_manifest_list
 
-        uri_for = lambda layer: self._storage.get_direct_download_url(
-            self._storage.locations, layer.blob.storage_path
-        )
+        def _join(first, second):
+            first.update(second)
+            return first
+
         body = {
             "hash": manifest.digest,
             "layers": [
                 {
                     "hash": str(l.layer_info.blob_digest),
-                    "uri": uri_for(l),
-                    "headers": {"Accept": ["application/gzip"],},
+                    "uri": self._blob_url_retriever.url_for_download(manifest.repository, l.blob),
+                    "headers": _join(
+                        {"Accept": ["application/gzip"],},
+                        (
+                            self._blob_url_retriever.headers_for_download(
```

Comment:
This is broken because Clair is expecting `headers` to be match the [type signature](https://godoc.org/net/http#Header) from the Go `http` package:
```go
type Header map[string][]string
```
So we need to send: 
```json
{
   "headers": {
       "Accept": ["application/gzip"],
       "Authorization": ["Bearer superlongauthtokenhere123"]
   }
}
```

Reply:
Fixed